meilisearch-http/src/index_controller/mod.rs

```diff
@@ -146,26 +159,23 @@ impl IndexController {
         settings: Settings,
         create: bool,
     ) -> anyhow::Result<UpdateStatus> {
-        let uuid = if create {
-            let uuid = self.uuid_resolver.get_or_create(uid).await?;
-            // We need to create the index upfront, since it would otherwise only be created when
-            // the update is processed. This would make calls to GET index to fail until the update
-            // is complete. Since this is get or create, we ignore the error when the index already
-            // exists.
-            match self.index_handle.create_index(uuid, None).await {
-                Ok(_) | Err(index_actor::IndexError::IndexAlreadyExists) => (),
-                Err(e) => return Err(e.into()),
-            }
-            uuid
-        } else {
-            self.uuid_resolver.resolve(uid).await?
+        let perform_udpate = |uuid| async move {
+            let meta = UpdateMeta::Settings(settings);
+            // Nothing so send, drop the sender right away, as not to block the update actor.
+            let (_, receiver) = mpsc::channel(1);
```

Comment:
so, are we sure this actually block the sender now?

Reply:
What do you mean?